### PR TITLE
br: continuing retry when pitr met the error about memory is limited

### PR DESCRIPTION
**Pull request by joccau**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/41983

Problem Summary:
- If the config `tikv.import.memory-use-ratio` is too low, `tikv-server` would report the error "memory is limited" to `BR` when PITR restore, and the PITR restore failed finally because the retry times exceeds the threshold.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Fix the issue that failed to restore log with the error "memory is limited".
```

### CODE REVIEW

File: br/pkg/restore/import_retry.go [hunk 1]
```diff
@@ -85,6 +86,13 @@ func (o *OverRegionsInRangeController) tryFindLeader(ctx context.Context, region
 
 // handleInRegionError handles the error happens internal in the region. Update the region info, and perform a suitable backoff.
 func (o *OverRegionsInRangeController) handleInRegionError(ctx context.Context, result RPCResult, region *split.RegionInfo) (cont bool) {
+	if result.StoreError.GetServerIsBusy() != nil {
+		if strings.Contains(result.StoreError.GetMessage(), "memory is limited") {
+			time.Sleep(15 * time.Second)
```
Comment (3pointer) [inline br/pkg/restore/import_retry.go:91]: shall we use fail point to speed up the test?

**Review by YuJuncen — APPROVED** [review]

**Review by 3pointer — APPROVED** [review]

**Pull request merged**